vcs/src/main/java/org/openjdk/skara/vcs/openjdk/CommitMessageParsers.java

```diff
@@ -116,6 +116,9 @@ public CommitMessage parse(List<String> lines) {
             Hash original = null;
             while (i < lines.size() && lines.get(i).equals("")) {
                 i++;
+                if (i >= lines.size()) {
+                    break;
+                }
```

Comment:
I guess I'm a little confused as to how this fixes the problem :/ Is it because `firstDelimiter` is false after the first line and then it adds a blank?

Does it make sense to move the `i++` to the end of the while loop, and then increment it one time separately before the loop, so that the while loop condition can check the `i >= lines.size()` condition, and no separate `if` is needed?

Reply:
I agree that the code is a bit unintuitive, have had to look at it a few times.. :) But no, incrementing should only be done if we're at a blank line, so can't do it unconditionally outside of the loop..

I pushed a different version that is perhaps a bit more clear.